### ISSUE TITLE
L.Point getCenter() getBounds(zoom)

### ISSUE THREAD

**Issue by syonfox** [Leaflet/Leaflet#7673]
**Is your feature request related to a problem? Please describe.**
So this might be a small thing but i think it should be a relatively simple as so just seeing what people think of it.
there should be a getCenter() method on the L.marker (points) and a getBounds(zoom) whitch calculates the point bounds at a zoom and returns those. (using current zoom if non provided). 

The problem this solves is cleaning up code by allowing logic to iterate over layers and not check weather getCenter Exists.
ideal there is some sort of standard layer interface that all layers have and programmers can amuse that a layer no mater the type has these utility functions. 

**Describe the solution you'd like**
For these methods to be added to L.circleMarker and L.marker and other point drawing classes

**Describe alternatives you've considered**
this can be solved but using the l.Marker.getlatlng function and constructing a bbox with that.  just pushes the implementation to the user.
basically its a inconvenience.  and I think adding these would be more intuitive and allow for cleaner code

```
featureGroup.eachLayer(l=>{
    let center = l.getCenter();
    //do whatever
})
//VS
featureGroup.eachLayer(l=>{
    let center
    if(typeof l.getCenter == 'function') {
        l.getCenter();
    } else {
        center = l.getLatLng(); 
    }
    //do whatever
})
```
and the bounds situation is slightly more complex. anyways just wondering if there is a reason for this i think it would be relatively simple to add.
a layer.getPixelBounds() function might be nice too but this might be redundant

**Comment by syonfox** [comment]
another ideas is a layer.getType function that returns the layers geojson type.
